prow/clonerefs/options.go

```diff
@@ -0,0 +1,177 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package clonerefs
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"flag"
+	"fmt"
+	"os"
+
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/test-infra/prow/kube"
+)
+
+// Options configures the clonerefs tool
+// completely and may be provided using JSON
+// or user-specified flags, but not both.
+type Options struct {
+	// SrcRoot is the root directory under which
+	// all source code is cloned
+	SrcRoot string `json:"src_root"`
+	// Log is the log file to which clone records are written
+	Log string `json:"log"`
+
+	// GitUserName is an optional field that is used with
```

Comment:
When do we not use this?

Reply:
The contract for `clonerefs` is that we either are initalizing a new repository or incrementally updating an existing one -- in the latter case we do not necessarily need to configure these/